Tigger/bin/tigger_convert.py

```diff
@@ -517,33 +522,41 @@ def convert_coordinates(coords):
                 parser.error("Invalid --add-brick setting %s" % brickspec)
             if [src.name for src in sources if src.name == srcname]:
                 print("Error: model already contains a source named '%s'" % srcname)
-            # add brick
-            from astropy.io import fits as pyfits
-            from astLib.astWCS import WCS
 
-            input_hdu = pyfits.open(fitsfile)[0]
+            input_hdu = fits.open(fitsfile)[0]
             hdr = input_hdu.header
             max_flux = float(input_hdu.data.max())
-            wcs = WCS(hdr, mode='pyfits')
-            # Get reference pixel coordinates
-            # wcs.getCentreWCSCoords() doesn't work, as that gives us the middle of the image
-            # So scan the header to get the CRPIX values
-            ra0 = dec0 = 1
-            for iaxis in range(hdr['NAXIS']):
-                axs = str(iaxis + 1)
-                name = hdr.get('CTYPE' + axs, axs).upper()
-                if name.startswith("RA"):
-                    ra0 = hdr.get('CRPIX' + axs, 1) - 1
-                elif name.startswith("DEC"):
-                    dec0 = hdr.get('CRPIX' + axs, 1) - 1
-            # convert pixel to degrees
-            ra0, dec0 = wcs.pix2wcs(ra0, dec0)
+
+            wcs, refpix, refsky, ra_axis, dec_axis = get_wcs_info(hdr)
+            naxis = hdr["NAXIS"]
+            if ra_axis is None or dec_axis is None:
+                print("Can't find RA and/or DEC axis in this FITS image")
+                sys.exit(1)
+
+            ra0, dec0 = refsky[ra_axis], refsky[dec_axis]
+
+            # convert reference pixel from degrees to rad
             ra0 *= DEG
             dec0 *= DEG
-            sx, sy = wcs.getHalfSizeDeg()
+            # get NX, NY
+            nx, ny = input_hdu.data.shape[-1:-3:-1]
+
+            # get half-size of image
+            refpix0 = refpix.copy()
+            refpix1 = refpix.copy()
+            refpix0[ra_axis], refpix0[dec_axis] = 0, 0
+            refpix1[ra_axis], refpix1[dec_axis] = nx, ny
+            refsky0 = wcs.wcs_pix2world(np.array(refpix0).reshape(1, naxis), 0)[0]
+            refsky1 = wcs.wcs_pix2world(np.array(refpix1).reshape(1, naxis), 0)[0]
+            ra0, dec0 = refsky0[ra_axis], refsky0[dec_axis]
+            ra1, dec1 = refsky1[ra_axis], refsky1[dec_axis]
+            sx, sy = abs(ra1 - ra0)/2, abs(dec1 - dec0) / 2
+            sx *= np.cos((dec1 + dec0)/2 * DEG)
```

Comment:
Not sure I quite follow here. Given sufficient image size, especially at declinations closer to the pole taking the RA extent at the fiducial point won't give you the true bounding box at the bottom (or top if near the SCP) of the image. I think this needs further thought. Should we not compute wcs extents at the top and bottom and place a bounding box around those?